x/msgfees/types/msgs_test.go

```diff
@@ -77,3 +79,298 @@ func TestMsgAssessCustomMsgFeeValidateBasic(t *testing.T) {
 		})
 	}
 }
+
+func TestMsgAddMsgFeeProposalRequestValidateBasic(t *testing.T) {
+	authority := sdk.AccAddress("input111111111111111").String()
+
+	cases := []struct {
+		name     string
+		msg      *MsgAddMsgFeeProposalRequest
+		errorMsg string
+	}{
+		{
+			"Empty type error",
+			NewMsgAddMsgFeeProposalRequest("", sdk.NewCoin("hotdog", sdk.NewInt(10)), "", "", authority),
+			"msg type is empty",
+		},
+		{
+			"Invalid fee amounts",
+			NewMsgAddMsgFeeProposalRequest("msgType", sdk.NewCoin("hotdog", sdk.NewInt(0)), "", "", authority),
```

Comment:
Relatedly, these tests are significantly easier to maintain if you don't use the `New...` constructor to make the thing you're testing. When a field is added to a message, and therefore the constructor, you then have to go back through all the tests and update the constructor call. But usually, when adding a field like that, all of the existing unit tests are still valid and wouldn't need to be changed if they didn't use the constructor.

Reply:
I will change them.  However, I do like having the constructor for that reason of breaking test or anywhere it is used.  It forces me to review the tests and update accordingly or remove ones that might be invalid after the updated constructor.  I was usually just using the fields when I needed to create the object in a state the constructor didn't allow.  I will start using the fields now going forward, just wanted to clarify my reasoning behind it.

Reply:
If the test itself doesn't break, we shouldn't have to update the code for it.